webpages/js/metaanalyses.js

```diff
@@ -2149,11 +2170,81 @@
     }
   }
 
+  function getFirstNonHiddenExp(paper) {
+    for (var i=0; i<paper.experiments.length; i++) {
+      if (!isHiddenExp(paper.experiments[i].id)) {
+        return i;
+      }
+    }
+
+    return 0;
+  }
+
+  function getNbUnhiddenExpPaper(paper) {
+    var retval = 0;
+
+    paper.experiments.forEach(function (exp) {
+      if (!isHiddenExp(exp.id)) {
+        retval += 1;
+      }
+    });
+
+    return retval;
+  }
+
+  function getNbHiddenExpPaper(paper) {
+    var retval = 0;
+
+    paper.experiments.forEach(function (exp) {
+      if (isHiddenExp(exp.id)) {
+        retval += 1;
+      }
+    });
+
+    return retval;
+  }
+
+  function isHiddenExp(expId) {
+    return currentMetaanalysis.hiddenExperiments.indexOf(expId) !== -1;
+  }
+
+  function unhideExpPaper(e) {
+    var paperIndex = 0;
+
+    while (currentMetaanalysis.papers[paperIndex].id != e.target.dataset.paperId) {
+      paperIndex++;
+    }
+
+    currentMetaanalysis.papers[paperIndex].experiments.forEach(function (experiment) {
+      var expIndex = currentMetaanalysis.hiddenExperiments.indexOf(experiment.id);
+      if (expIndex != -1) {
+        currentMetaanalysis.hiddenExperiments.splice(expIndex, 1);
+      }
+    });
+
+    updateMetaanalysisView();
+    _.scheduleSave(currentMetaanalysis);
+  }
+
+  function hideExp(e) {
+    currentMetaanalysis.hiddenExperiments.push(e.target.dataset.id);
+    unpinPopupBox();
+    updateMetaanalysisView();
+    _.scheduleSave(currentMetaanalysis);
+  }
+
+  function addExpUnhideButton(expNode, paperId) {
+    var unhidebutton = _.findEl(expNode, '.unhideexp');
+    unhidebutton.removeAttribute('hidden');
+    _.setDataProps(expNode, 'button.unhideexp', 'paperId', paperId);
+    _.addEventListener(expNode, 'button.unhideexp', 'click', unhideExpPaper);
+  }
+
   function addPaperRow() {
     lima.requestPaper('new-paper')
     .then(function (newPaper) {
       // Populate an empty experiment
-      newPaper.experiments.push({});
+      newPaper.experiments.push({id: '/id/exp/' + Date.now()});
```

Comment:
Sorry, Papers use `tools.uniqueNow()`, presumably we care less although we do need to cater for experiments that do not have id's currently, i.e. migrating.

@jacekkopecky - We talked about experiment ids being client side generated, so here seems like a good place for this. Do you have a different idea? 

I am also assuming migration can go in storage.js (both in `migratePaper` and `migrateMetaanalysis`?) as something like;

```
paper.experiments.forEach(function(experiment) {
  if (!experiment.id) {
    experiment.id = '/id/exp/' + Date.now();
  }
});
```